src/logging.jl

```diff
@@ -0,0 +1,39 @@
+import Logging, LoggingExtras
+
+export MPILogger, MPIFileLogger
+
+"""
+    MPILogger(iostream = stdout)
+    
+Add a rank prefix before log messages.
+
+Takes in `stdout` as a default IOStream.
+"""
+function MPILogger(ctx::AbstractCommsContext, iostream = stdout)
```

Comment:
```suggestion
MPILogger(ctx::AbstractCommsContext) = MPILogger(stdout, ctx)
function MPILogger(iostream, ctx::AbstractCommsContext)
```
I think it's somewhat common for `io` streams to be the first argument (perhaps for do-like syntax), we could write it like this.